device-server/src/main/kotlin/com/badoo/automation/deviceserver/host/SimulatorsNode.kt

```diff
@@ -102,6 +102,24 @@ class SimulatorsNode(
         }
     }
 
+    override fun getDiagnostic(deviceRef: DeviceRef, type: DiagnosticType): Diagnostic {
+        when (type) {
+            DiagnosticType.SystemLog -> return Diagnostic(
+                type = type,
+                content = getDeviceFor(deviceRef).systemLog.content()
+            )
+            else -> throw  RuntimeException("Diagnostic $type is not supported")
+        }
+    }
+
+    override fun resetDiagnostic(deviceRef: DeviceRef, type: DiagnosticType) {
+        @Suppress("REDUNDANT_ELSE_IN_WHEN")
```

Comment:
better not suppress. Kotlin compiler will tell, when a new DyagnosticType is added. Let's remove else